### PR TITLE
feat(523): Add REMOVE route for Collections

### PR DESCRIPTION
**Pull request by joelseq**
## Context

For collections, we need to be able to delete a collection.

## Objective

This PR adds a REMOVE route for collections so that you can delete a collection.

## References

Issue: #523

### CODE REVIEW

File: plugins/collections/remove.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    method: 'DELETE',
+    path: '/collections/{id}',
+    config: {
+        description: 'Delete a single collection',
+        notes: 'Returns null if successful',
+        tags: ['api', 'collections'],
+        auth: {
+            strategies: ['token', 'session'],
+            scope: ['user']
+        },
```
Comment (ian-fox) [inline plugins/collections/remove.js]: hapi-swagger block?

File: plugins/collections/remove.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                userFactory.get({ username })
+            ]).then(([collection, user]) => {
+                if (!collection) {
+                    throw boom.notFound('Collection does not exist');
+                }
+                if (!user) {
+                    throw boom.notFound(`User ${username} does not exist`);
+                }
+                if (collection.userId !== user.id) {
+                    throw boom.unauthorized(`User ${username} does not have permission`);
```
Comment (ian-fox) [inline plugins/collections/remove.js]: Maybe `does not own collection`?

File: test/plugins/collections.test.js [hunk 3]
```diff
@@ -368,4 +368,60 @@ describe('collection plugin test', () => {
...
+        });
+
+        it('returns 204 when delete is successful', () =>
+            server.inject(options).then((reply) => {
+                assert.equal(reply.statusCode, 204);
+                assert.calledOnce(collectionMock.remove);
+            })
+        );
+
+        it('returns 401 when user does not have permiddion', () => {
```
Comment (minz1027) [inline test/plugins/collections.test.js]: permiddion? LOL

File: plugins/collections/remove.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const { collectionFactory, userFactory } = request.server.app;
+            const { username } = request.auth.credentials;
+
+            // Fetch the colleciton and user models
```
Comment (FenrirUnbound) [inline plugins/collections/remove.js]: collection*

File: test/plugins/collections.test.js [hunk 5]
```diff
@@ -368,4 +368,60 @@ describe('collection plugin test', () => {
...
+            const fakeUserId = 12;
+
```
Comment (FenrirUnbound) [inline test/plugins/collections.test.js]: Extra whitespace

File: plugins/collections/remove.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            return Promise.all([
+                collectionFactory.get(request.params.id),
+                userFactory.get({ username })
+            ]).then(([collection, user]) => {
+                if (!collection) {
```
Comment (FenrirUnbound) [inline plugins/collections/remove.js]: This case is not covered

File: test/plugins/collections.test.js (hunk 5 above)
Comment (FenrirUnbound) [inline test/plugins/collections.test.js]: For clarification, this comment is not a big deal; you can leave it.

File: test/plugins/collections.test.js (hunk 3 above)
Comment (joelseq) [inline test/plugins/collections.test.js]: This is weird... I fixed that typo before and pushed it earlier but it's not updating on here? It's permission on my local

**Pull request merged**